compiler/src/yul/mappers/expressions.rs

```diff
@@ -167,6 +167,22 @@ pub fn expr_bin_operation(
     unreachable!()
 }
 
+pub fn expr_unary_operation(
+    context: &Context,
+    exp: &Spanned<fe::Expr>,
+) -> Result<yul::Expression, CompileError> {
+    if let fe::Expr::UnaryOperation { op, operand } = &exp.node {
+        let yul_operand = expr(context, operand)?;
+        if let fe::UnaryOperator::USub = &op.node {
+            let zero = literal_expression! {0};
+            // TODO: Is there no way to just negate yul_operand?
+            return Ok(expression! { sub([zero], [yul_operand]) });
```

Comment:
@g-r-a-n-t  Maybe I'm doing it wrong. I've fiddled quite a while with this trying to find a way to just negate `[yul_operand]`. At the end I settled with the `sub(0, <expr>)` operation but it feels like a hack to me. Am I doing it wrong?

Reply:
I think this makes the most sense

Reply:
But doesn't this waste gas for the `sub` operation? Shouldn't we be able to just turn `6` into `-6` and `foo` into `-foo`? Is this a limitation of the shorthand macros or yul itself? I guess I could take a snippet of solidity and run it through `--experimental-via-ir` to find out. Same goes for the division case.

Reply:
I don't believe Yul has a way to represent negative numbers in the language itself, so there isn't a way for us to write `-6` or `-foo` in Yul. This behavior needs to be implemented by the Fe compiler on top of Yul's `u256` type. 

If we were to assume that negative signs may only be applied to literals (`-6` and not `-foo`), then we could potentially map this value to its correct binary value using an unsigned value. So, `-6` would become `2475880078570760549798248447` in the Yul code. This would save a small amount of gas, but it makes the Yul code more difficult to read.

The other option is to do what you already have and map `-6` to `sub(0, 6)`, which I think makes the Yul code more readable.

Reply:
I was just reading the Yul docs again and saw that they have a few built in functions for signed operations: `sdiv`, `smod`, `slt`, `sgt`, and `sar`.

https://docs.soliditylang.org/en/latest/yul.html

Reply:
Oh! Nice find. That should help :)

Reply:
Oh and btw, solidity negates numbers the same way: `sub(0, _value)`. In other news `solc foo.sol  --ir` is becoming my new swiss army knife :switzerland: